File: app/src/main/java/com/bas080/autosleepdroid/AppDefaults.java
package com.bas080.autosleepdroid;

public class AppDefaults {
    public static final int DURATION_MINUTES = 20;
    public static final int MINUTES_MIN = 1;
    public static final int MINUTES_MAX = 24 * 60; // 1440

    public static final int WAKE_UP_GOAL_HOUR = 6;
    public static final int WAKE_UP_GOAL_MINUTE = 30;

    public static final int MIN_SLEEP_DURATION_MINUTES = 450; // 7h 30m
    public static final int HC_MIN_DURATION_MINUTES = 15;
    public static final int NAP_DURATION_MINUTES = 20;

    public static final long FADE_DURATION_MS = 30_000L;
    public static final long FADE_STEP_INTERVAL_MS = 1_000L;
    public static final int TOTAL_FADE_STEPS = (int) (FADE_DURATION_MS / FADE_STEP_INTERVAL_MS);

    public static final long SNOOZE_DURATION_MS = 9 * 60_000L;
    public static final long ALARM_CRESCENDO_DURATION_MS = 3 * 60_000L;
    public static final long ALARM_CRESCENDO_INTERVAL_MS = 500L;
}

File: app/src/main/java/com/bas080/autosleepdroid/AppDefaults.kt
package com.bas080.autosleepdroid

object AppDefaults {
    const val DURATION_MINUTES: Int = 20
    const val MINUTES_MIN: Int = 1
    const val MINUTES_MAX: Int = 24 * 60 // 1440

    const val WAKE_UP_GOAL_HOUR: Int = 6
    const val WAKE_UP_GOAL_MINUTE: Int = 30

    const val MIN_SLEEP_DURATION_MINUTES: Int = 450 // 7h 30m
    const val HC_MIN_DURATION_MINUTES: Int = 15

    const val FADE_DURATION_MS: Long = 30_000L
    const val FADE_STEP_INTERVAL_MS: Long = 1_000L
    const val TOTAL_FADE_STEPS: Int = (FADE_DURATION_MS / FADE_STEP_INTERVAL_MS).toInt()

    const val SNOOZE_DURATION_MS: Long = 9 * 60_000L
    const val ALARM_CRESCENDO_DURATION_MS: Long = 3 * 60_000L
    const val ALARM_CRESCENDO_INTERVAL_MS: Long = 500L
}
